addons/web/static/tests/core/dropdown_tests.js

```diff
@@ -1146,4 +1146,26 @@ QUnit.module("Components", ({ beforeEach }) => {
             assert.notOk(target.querySelector(".o_datepicker_input").value === "");
         }
     );
+
+    QUnit.test("OnOpened callback props called after the menu has been mounted", async (assert) => {
+        class Parent extends Component {
+            beforeOpenCallback() {
+                assert.step("beforeOpened");
```

Comment:
make a deferred and await it here after the assert.step (there is a makeDeferred test util)